src/copy.rs

```diff
@@ -107,6 +107,20 @@ where
     Ok(())
 }
 
+// During shutdown, the other end may have already disconnected. That is fine, they shutdown for us.
+// Ignore it.
+fn ignore_shutdown_errors(res: Result<(), io::Error>) -> Result<(), io::Error> {
+    match &res {
+        Err(e)
+            if e.kind() == io::ErrorKind::NotConnected
+                || e.kind() == io::ErrorKind::UnexpectedEof =>
+        {
+            Ok(())
```

Comment:
Would we still get a `trace` in this case? Should probably still be able to see this is being ignored @ `trace` level.

Reply:
good idea; done